src/components/document-list/_macro.njk

```diff
@@ -0,0 +1,114 @@
+        {% for document in (params.documents if params.documents is iterable else params.documents.items()) %}
+
+        <li class="ons-document-list__item{{ ' ons-document-list__item--featured' if document.featured is defined and document.featured }}{{ ' ons-document-list__item--full-width' if document.fullWidth is defined and document.fullWidth == true }}{{ ' ' + document.classes if document.classes is defined and document.classes else '' }}"{% if document.attributes is defined and document.attributes %}{% for attribute, value in (document.attributes.items() if document.attributes is mapping and document.attributes.items else document.attributes) %} {{attribute}}="{{value}}"{% endfor %}{% endif %}>
+
+            {% if document.fullWidth is defined and document.fullWidth == true %}
+
+                <div class="ons-container{{ ' ons-container--wide' if document.wide is defined and document.wide == true }}">
+
+            {% endif %}
+
+                    {% if document.thumbnail is defined or document.thumbnail == true %}
+
+                        <div class="ons-document-list__item-image{%- if document.metadata.file is defined and document.metadata.file %} ons-document-list__item-image--file{% endif %}" aria-hidden="true">
+                            <a class="ons-document-list__image-link{{ ' ons-document-list__image-link--placeholder' if not document.thumbnail.smallSrc }}" href="{{ document.url }}" tabindex="-1">
+                                {% if document.thumbnail.smallSrc is defined and document.thumbnail.smallSrc and document.thumbnail.largeSrc is defined and document.thumbnail.largeSrc %}
+                                    <img srcset="{{ document.thumbnail.smallSrc }} 1x, {{ document.thumbnail.largeSrc }} 2x" src="{{ document.thumbnail.smallSrc }}" alt="" loading="lazy">
+                                {% endif %}
+                            </a>
+                        </div>
+
+                    {% endif %}
+
+                    <div class="ons-document-list__item-content">
+
+                        <div class="ons-document-list__item-header{{ ' ons-document-list__item-header--reverse' if document.showMetadataFirst is defined and document.showMetadataFirst == true }}">
+
+                            {% set titleTag = params.titleTag | default("h2") %}
+
+                            <{{ titleTag }} class="ons-document-list__item-title ons-u-fs-m ons-u-mt-no ons-u-mb-xs">
+                                <a href="{{ document.url }}">{{ document.title }}
+                                {%- if document.metadata.file is defined and document.metadata.file %}
```

Comment:
Need an if around this section including the fileType stuff that checks if `document.metadata` exists